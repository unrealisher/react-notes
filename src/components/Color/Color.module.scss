.color {
  cursor: pointer;
  display: inline-block;
  width: 48px;
  height: 48px;
  border-radius: 4px;
  opacity: 0.4;

  &:hover,
  &:focus {
    opacity: 0.8;
  }

  &:active {
    opacity: 0.6;
  }

  @media (min-width: 516px) {
    width: 32px;
    height: 32px;
  }
}
