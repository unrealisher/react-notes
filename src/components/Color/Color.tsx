import React from "react";

import styles from "./Color.module.scss";
import IColor from "../../interfaces/IColor";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { setFilterItems } from "../../store/actions/filterItems";

interface IDispatchToProps {
  onFilterChange: Function;
}

interface Props extends IDispatchToProps {
  color: IColor;
}

const Color = (props: Props): JSX.Element => {
  const { color, onFilterChange } = props;
  return (
    <React.Fragment>
      <input
        className={styles.checkbox}
        type="checkbox"
        id={`checkbox_${color.color}`}
        onChange={() => onFilterChange(color.id)}
        hidden
      />
      <label
        className={styles.color}
        style={{ backgroundColor: color.color }}
        htmlFor={`checkbox_${color.color}`}
      />
    </React.Fragment>
  );
};

const mapDispatchToProps = (dispatch: Dispatch): IDispatchToProps => {
  return {
    onFilterChange: (id: number): void => {
      dispatch(setFilterItems(id));
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(Color);
