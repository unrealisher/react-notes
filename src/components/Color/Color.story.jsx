import React from "react";

import Color from "./Color";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

storiesOf("Color", module).add("Красный", () => <Color color="red" />);
