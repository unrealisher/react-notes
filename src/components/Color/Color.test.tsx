import React from "react";
import { shallow } from "enzyme";
import { Color } from "./Color";

describe("Color", () => {
  const color = { id: 0, color: "#E84747" };
  it("correct render", () => {
    const component = shallow(
      <Color color={color} filter={[]} onFilterChange={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
