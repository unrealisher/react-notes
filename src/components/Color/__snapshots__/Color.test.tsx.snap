// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Color
    color={
      Object {
        "color": "#E84747",
        "id": 0,
      }
    }
    filter={Array []}
    onFilterChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          className="checkbox"
          hidden={true}
          id="checkbox_#E84747"
          onChange={[Function]}
          type="checkbox"
        />,
        <label
          className="color"
          htmlFor="checkbox_#E84747"
          style={
            Object {
              "backgroundColor": "#E84747",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "checkbox",
          "hidden": true,
          "id": "checkbox_#E84747",
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "color",
          "htmlFor": "checkbox_#E84747",
          "style": Object {
            "backgroundColor": "#E84747",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            className="checkbox"
            hidden={true}
            id="checkbox_#E84747"
            onChange={[Function]}
            type="checkbox"
          />,
          <label
            className="color"
            htmlFor="checkbox_#E84747"
            style={
              Object {
                "backgroundColor": "#E84747",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "checkbox",
            "hidden": true,
            "id": "checkbox_#E84747",
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "color",
            "htmlFor": "checkbox_#E84747",
            "style": Object {
              "backgroundColor": "#E84747",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
